app/src/main/kotlin/com/wire/android/ui/userprofile/UserProfileViewModel.kt

```diff
@@ -31,32 +29,69 @@ import kotlinx.coroutines.withContext
 class UserProfileViewModel @Inject constructor(
     private val navigationManager: NavigationManager,
     private val dataStore: UserDataStore,
-    private val uploadUserAvatar: UploadUserAvatarUseCase,
+    private val getPublicAssetUseCase: GetPublicAssetUseCase,
     private val getSelf: GetSelfUserUseCase
 ) : ViewModel() {
 
     var userProfileState by mutableStateOf(SelfUserProfileState())
         private set
 
     init {
-        // TODO: here we should have a loading state as the first initial state of the screen
         viewModelScope.launch {
-            getSelf().collect {
-                userProfileState = SelfUserProfileState(
-                    status = UserStatus.AVAILABLE,
-                    fullName = it.name.orEmpty(),
-                    userName = it.handle.orEmpty(),
-                    teamName = it.team,
-                    // Add some mocked team
-                    otherAccounts = listOf(
-                        OtherAccount("someId", "", it.name.orEmpty(), "Wire Swiss GmbH"),
-                        OtherAccount("someId", "", "B. A. Baracus", "The A-Team"),
+            fetchSelfUser()
+        }
+    }
+
+    private suspend fun fetchSelfUser() {
+        // Show progress bar on initial load
+        showLoadingAvatar(true)
+
+        viewModelScope.launch {
+            getSelf().collect { selfUser ->
+                with(selfUser) {
+                    // Load user avatar raw image data
+                    val avatarAssetRaw = completePicture?.let { getAvatarAsByteArrayOrNull(it) }
+
+                    // Update avatar asset id on user data store
+                    completePicture?.apply { dataStore.updateUserAvatarAssetId(this) }
```

Comment:
The idea was simply to cache the asset id on the data store and that way save one more call to the `getSelf` api and only make a call to the `GetPublicAssetUseCase` which will return it from memory in case it had been also cached previously

Reply:
I just feel like maybe we should rather cache it on the repository level, it would be easier to use - what do you think?

Reply:
But this is already happening with `downloadPublicAsset` method, isn't it? But we still need to keep somewhere a reference/id to the specific asset we cached previously and that we want to retrieve again

Reply:
Or you meant, also adding methods to save/load the avatar asset id, or directly the self user avatar?

Reply:
Yeah, can't we also cache this asset id in the repository? 😄

Reply:
In any case I will mark it with a TODO: to adapt to this once we revisit this for the File sharing feature, where we will most likely create another db table to store the assets ids